File: app/src/main/java/com/cangwang/modulebus/ExModule/ModuleExFragment.java
package com.cangwang.modulebus.ExModule;

import android.os.Bundle;

import com.cangwang.core.ModuleBus;
import com.cangwang.core.cwmodule.ex.ModuleManageExFragment;

import java.util.ArrayList;
import java.util.List;

/**
 * Created by cangwang on 2017/6/15.
 */
public class ModuleExFragment extends ModuleManageExFragment {
    public static ModuleExFragment newInstance(){
        ModuleExFragment fragment = new ModuleExFragment();
        Bundle bundle = new Bundle();
        fragment.setArguments(bundle);
        return fragment;
    }

    public static ModuleExFragment newInstance(List<String> modules){
        PageExConfig.moduleList = modules;
        ModuleExFragment fragment = new ModuleExFragment();
        Bundle bundle = new Bundle();
        fragment.setArguments(bundle);
        return fragment;
    }


    @Override
    public String moduleConfig() {
        return "normal";
    }
}

File: app/src/main/java/com/cangwang/modulebus/ExModule/ModuleExFragment.kt
package com.cangwang.modulebus.ExModule

import android.os.Bundle
import com.cangwang.core.cwmodule.ex.ModuleManageExFragment

/**
 * Created by cangwang on 2017/6/15.
 */
class ModuleExFragment : ModuleManageExFragment() {
    override fun moduleConfig(): String {
        return "normal"
    }

    companion object {
        fun newInstance(): ModuleExFragment {
            val fragment = ModuleExFragment()
            val bundle = Bundle()
            fragment.arguments = bundle
            return fragment
        }

        fun newInstance(modules: List<String?>?): ModuleExFragment {
            PageExConfig.moduleList = modules
            val fragment = ModuleExFragment()
            val bundle = Bundle()
            fragment.arguments = bundle
            return fragment
        }
    }
}